File: app/src/androidTest/java/com/uramonk/androidtemplateapp/repository/WeatherRepositoryTest.java
package com.uramonk.androidtemplateapp.repository;

import android.support.test.runner.AndroidJUnit4;

import com.uramonk.androidtemplateapp.api.WeatherApi;
import com.uramonk.androidtemplateapp.entity.WeatherEntity;
import com.uramonk.androidtemplateapp.module.NetworkModule;
import com.uramonk.androidtemplateapp.module.WeatherModule;
import com.uramonk.androidtemplateapp.utility.TestUtility;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.MockitoAnnotations;

import java.io.IOException;

import okhttp3.OkHttpClient;
import okhttp3.mockwebserver.MockResponse;
import okhttp3.mockwebserver.MockWebServer;

import retrofit2.Retrofit;

import static org.junit.Assert.assertThat;
import static org.hamcrest.CoreMatchers.is;

/**
 * Created by uramonk on 2016/07/02.
 */
@RunWith(AndroidJUnit4.class)
public class WeatherRepositoryTest {
    @Before
    public void setUp() {
        MockitoAnnotations.initMocks(this);
    }

    private IWeatherRepository getWeatherRepository(MockWebServer mockWebServer) {
        Retrofit retrofit = TestUtility.getRetrofitWithMockWebServer(mockWebServer);

        WeatherModule weatherModule = new WeatherModule();
        WeatherApi weatherApi = weatherModule.provideWeatherApi(retrofit);
        return weatherModule.provideWeatherRepository(weatherApi);
    }

    @Test
    public void Weatherを取得できる() throws IOException {
        final MockWebServer mockWebServer = new MockWebServer();
        IWeatherRepository weatherRepository = getWeatherRepository(mockWebServer);
        mockWebServer.enqueue(new MockResponse().setBody(TestUtility.createJsonString("weather_normal.json")));

        WeatherEntity weatherEntity = weatherRepository.getWeather().toBlocking().single();
        assertThat(weatherEntity.base, is("stations"));
        assertThat(weatherEntity.weather.size(), is(1));
        assertThat(weatherEntity.weather.get(0).description, is("broken clouds"));
        assertThat(weatherEntity.weather.get(0).icon, is("04d"));
        assertThat(weatherEntity.weather.get(0).id, is(803));
        assertThat(weatherEntity.weather.get(0).main, is("Clouds"));

        mockWebServer.shutdown();
    }
}

File: app/src/androidTest/java/com/uramonk/androidtemplateapp/repository/WeatherRepositoryTest.kt
package com.uramonk.androidtemplateapp.repository

import android.support.test.runner.AndroidJUnit4

import com.uramonk.androidtemplateapp.api.WeatherApi
import com.uramonk.androidtemplateapp.entity.WeatherEntity
import com.uramonk.androidtemplateapp.module.WeatherModule
import com.uramonk.androidtemplateapp.utility.TestUtility

import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.MockitoAnnotations

import java.io.IOException

import okhttp3.mockwebserver.MockResponse
import okhttp3.mockwebserver.MockWebServer
import retrofit2.Retrofit

import org.hamcrest.CoreMatchers.`is`
import org.junit.Assert.assertThat

/**
 * Created by uramonk on 2016/07/02.
 */
@RunWith(AndroidJUnit4::class)
class WeatherRepositoryTest {
    @Before
    fun setUp() {
        MockitoAnnotations.initMocks(this)
    }

    private fun getWeatherRepository(mockWebServer: MockWebServer): IWeatherRepository {
        val retrofit = TestUtility.getRetrofitWithMockWebServer(mockWebServer)

        val weatherModule = WeatherModule()
        val weatherApi = weatherModule.provideWeatherApi(retrofit)
        return weatherModule.provideWeatherRepository(weatherApi)
    }

    @Test
    @Throws(IOException::class)
    fun Weatherを取得できる() {
        val mockWebServer = MockWebServer()
        val weatherRepository = getWeatherRepository(mockWebServer)
        mockWebServer.enqueue(MockResponse().setBody(TestUtility.createJsonString("weather_normal.json")))

        val weatherEntity = weatherRepository.getWeather().toBlocking().single()
        assertThat(weatherEntity.base, `is`("stations"))
        assertThat(weatherEntity.weather.size, `is`(1))
        assertThat(weatherEntity.weather[0].description, `is`("broken clouds"))
        assertThat(weatherEntity.weather[0].icon, `is`("04d"))
        assertThat(weatherEntity.weather[0].id, `is`(803))
        assertThat(weatherEntity.weather[0].main, `is`("Clouds"))

        mockWebServer.shutdown()
    }
}
